sympy/core/mul.py

```diff
@@ -86,12 +86,60 @@ def _unevaluated_Mul(*args):
 
 
 class Mul(Expr, AssocOp):
+    """
+    Multiplication operation for algebraic group.
+
+    This class mainly deals with the multiplication in the field of complex numbers.
+    Any abstract algebraic field can be dealt as well, but it is encouraged to
+    define specific class such as ``MatMul`` or ``VectorMul`` for other fields.
+    This class also serves as base class for these classes.
+
+    Another use of ``Mul()`` is to behave as template for abstract multiplication so
+    that its arguments can be substituted to return different class. Refer to
+    examples section for this.
+
+    Since multiplication can be vector space operation, arguments may have the different
+    kind. Kind of the resulting object is automatically inferred.
```

Comment:
Can we make a reference here that is clickable in the web docs?

Perhaps it would make sense to add a "see also" at the bottom that references Kind as well as Add and Pow.

It should also be mentioned that `Add` can only have `Expr` args.